### PR TITLE
`deleteFlyer` mutation

### PR DESCRIPTION
**Pull request by vinnie4k**
## Overview

This PR implements a mutation to delete a flyer.

### deleteFlyer Mutation

- Created a mutation that deletes a Flyer if successful given the following arguments:
    - {string} `id` the ID of the Flyer to delete
- This mutation returns the deleted flyer. If the flyer cannot be found, the mutation will throw an error.
- **Note that if the POST request to remove the flyer image from our servers fails, the Flyer will still be deleted in the MongoDB.**

### removeImage

- Created a function in `utils.ts` to remove an image from our servers, given the bucket name and image URL
- Created an interface `UploadResponse` with the properties `data: String` and `success: boolean` to store the parsed JSON response.

## Test Coverage
- I started by testing this mutation through the GraphQL playground with my local MongoDB:

```
mutation DeleteFlyer($id: String!) {
  deleteFlyer(id: $id) {
    ...flyerFields
  }
}
```

- To ensure that this mutation works as expected, I provided the following invalid arguments to see if the request fails:
    - Invalid `id`. See the first screenshot.
- The second screenshot shows a successful delete mutation. My local MongoDB is updated correctly.
- I then wrote test cases to test this mutation. See screenshot below.

## Next Steps
- Work on implementing editing flyer mutation.
- I believe that all functions should have proper documentation and the test suite description should also be refactored. Perhaps this could be something to do in the future.

## Screenshots
delete_flyer_1
delete_flyer_2
delete_flyer_3
delete_flyer_4

### CODE REVIEW

File: src/common/constants.ts
Comment (Archit404Error) [inline src/common/constants.ts:23]: This needs to be removed form the git history as well in accordance with #102, you should try to git reset your commits here and force a rebase onto your other branch once you've removed the sensitive information there. Will come back to this and do a full review once that's done.

**Review by Archit404Error — COMMENTED** [review]
Just suggesting this one change for now, will re-review once this is resolved

File: src/repos/FlyerRepo.ts
Comment (Archit404Error) [inline src/repos/FlyerRepo.ts:284]: Generally the `@function` is used in JSDoc when we want to specify an object is a function if it is unclear (e.g., if we did `let x = someFunc()` and someFunc() returns a function, we would mark x with @function. I don't think we need this here

File: src/utils.ts
Comment (Archit404Error) [inline src/utils.ts:42]:
```suggestion
  const responseData = await response.json();
```

File: src/utils.ts
Comment (Archit404Error) [inline src/utils.ts:34]: Once again would recommend moving this upload bucket variable into the env file as well and purging from git history

**Review by Archit404Error — CHANGES_REQUESTED** [review]
Essentially good to go, I want you to move the upload bucket into the env file in #102 and then rebase this branch onto that one, purging the upload bucket from the constants file as well since it's good to keep the buckets secret.

Once that's done I'll approve

File: src/utils.ts
Comment (vinnie4k) [reply src/utils.ts:41]: I tried doing this before but the issue is that I want to access the `success` propery of the JSON. Using `response.json()` will give `responseData` an unknown type so Typescript will say "Property `success` does not exist on type `unknown`".

Comment (Archit404Error) [reply src/utils.ts:41]: Yes what you should do then is define a type that specifies what the response JSON will look like and specify that that is the type of the JSON response data. You can checkout an example of this in one of my repos. I define the type interface [here](https://github.com/Archit404Error/SemanticClassSearch/blob/main/mobile/index.d.ts) and make the request [here](https://github.com/Archit404Error/SemanticClassSearch/blob/8d1db8c34490bfbe428080ff4a43802640e54d5c/mobile/app/(tabs)/search/index.tsx#L9-L14)

Comment (vinnie4k) [reply src/utils.ts:41]: I created an interface at the top of `utils.ts` called `UploadResponse` and wrote the following:
```
const responseData = (await response.json()) as UploadResponse;
if (!responseData.success) {
  console.log(`Removing an image from our servers failed with image URL: ${imageURL}`);
}
return responseData.success;
```
This works as intended, but do you want to have a `.d.ts` file instead?

Comment (Archit404Error) [reply src/utils.ts:41]: The interface should be fine for now, we can always pull this out into a .d.ts file later

Comment (Archit404Error) [reply src/utils.ts:41]: You might also be able to just do `const responseData : UploadResponse = await response.json()`. Once you commit this I'll approve

**Review by Archit404Error — COMMENTED** [review]
Will approve once you add in the interface, try doing `const resJson : Interface = await response.json()` and if that doesn't work you can use the existing code you have

File: src/utils.ts
Comment (vinnie4k) [reply src/utils.ts]: `const responseData : UploadResponse = await response.json()` doesn't work. I believe it's because the response has an `undefined` type. I kept what I wrote earlier and everything works perfectly for adding and deleting flyers.

File: src/repos/FlyerRepo.ts
Comment (Archit404Error) [inline src/repos/FlyerRepo.ts:223]: Remove this

**Review by Archit404Error — APPROVED** [review]
Good to go, just remove the extraneous `@function` in your JSDoc

**Pull request merged**
